File: tools/list_files.lua
local luapython = require("luapython")

return {
    name = "list_files",
    description = "List all files in the current directory",
    tool = {
        type = "function",
        ["function"] = {
            name = "list_files",
            description = "List all files in the current directory",
            parameters = {
                type = "object",
                properties = {},
                required = luapython.list {}
            }
        }
    },
    action = function(input)
        local files_content
        local p = io.popen("ls -R")
        if not p then
            return { success = false, content = "Failed to list files" }
        end
        files_content = p:read("*a")
        p:close()
        return { success = true, content = files_content }
    end
}

File: tools/list_files.py
"""list_files tool — list all files recursively."""

name = "list_files"
description = "List all files in the current directory recursively"
schema = {
    "type": "function",
    "function": {
        "name": "list_files",
        "description": "List all files in the current directory recursively",
        "parameters": {
            "type": "object",
            "properties": {},
            "required": [],
        },
    },
}


def action(input: dict) -> dict:
    import subprocess
    try:
        result = subprocess.run(["ls", "-R"], capture_output=True, text=True, timeout=10)
        return {"success": True, "content": result.stdout}
    except Exception as e:
        return {"success": False, "content": f"Failed to list files: {e}"}
